conda_build/variants.py

```diff
@@ -748,12 +748,22 @@ def find_used_variables_in_text(variant, recipe_text, selectors_only=False):
         conditional_regex = (
             r"(?:^|[^\{])\{%\s*(?:el)?if\s*.*" + v_regex + r"\s*(?:[^%]*?)?%\}"
         )
+        for_regex = r"(?:^|[^\{])\{%\s*for\s*.*\s*in\s*" + v_regex + r"(?:[^%]*?)?%\}"
```

Comment:
Apparently Jinja supports things like:

```jinja
{% for user in users if not user.hidden %}
    <li>{{ user.username|e }}</li>
{% endfor %}
```

or 

```jinja
<ul class="sitemap">
{%- for item in sitemap recursive %}
    <li><a href="{{ item.href|e }}">{{ item.title }}</a>
    {%- if item.children -%}
        <ul class="submenu">{{ loop(item.children) }}</ul>
    {%- endif %}</li>
{%- endfor %}
</ul>
```

, which this regex would not over I think? But  I've never seen it in the wild, so maybe it's not worth implementing here. A `TODO` comment might be in order.